driver-service/src/test/java/com/modsen/driverservice/controller/impl/CarControllerTest.java

```diff
@@ -1,123 +1,174 @@
-        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders.delete(CAR_DELETE_ENDPOINT, CAR_ID);
+        doNothing().when(carService).safeDeleteCarByCarId(anyLong());
+        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders
+                .delete(AppTestUtil.CAR_DELETE_ENDPOINT, AppTestUtil.CAR_ID);
+
+        // Act and Assert
+        mockMvc.perform(requestBuilder)
+                .andExpect(MockMvcResultMatchers.status().isNoContent());
+    }
+
+    @Test
+    @DisplayName("Test safeDeleteCarById(Long); then returns NOT_FOUND status code")
+    void testSafeDeleteCarById_thenReturnsNotFoundStatusCode_whenSuchCarDoesntExists() throws Exception {
+        //Arrange
+        doThrow(new CarNotFoundException("")).when(carService).safeDeleteCarByCarId(anyLong());
 
-        // Act
-        ResultActions actualPerformResult = mockMvc.perform(requestBuilder);
+        MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders
+                .delete(AppTestUtil.CAR_DELETE_ENDPOINT, AppTestUtil.CAR_ID);
 
-        // Assert
-        actualPerformResult.andExpect(MockMvcResultMatchers.status().isNoContent());
+        // Act and Assert
+        mockMvc.perform(requestBuilder)
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
     }
 
     @Test
-    @DisplayName("Test getCarById(Long)")
-    void testGetCarById() throws Exception {
+    @DisplayName("Test getCarById(Long); then success")
+    void testGetCarById_thenSuccess() throws Exception {
         // Arrange
-        when(carService.getCarById(Mockito.<Long>any()))
-                .thenReturn(createCarDto());
+        when(carService.getCarById(anyLong()))
+                .thenReturn(AppTestUtil.carResponseDto);
 
         MockHttpServletRequestBuilder requestBuilder = MockMvcRequestBuilders
-                .get(CAR_GET_ENDPOINT, CAR_ID);
+                .get(AppTestUtil.CAR_GET_ENDPOINT, AppTestUtil.CAR_ID);
 
         // Act and Assert
         mockMvc.perform(requestBuilder)
                 .andExpect(MockMvcResultMatchers.status().isOk())
-                .andExpect(MockMvcResultMatchers.content().contentType("application/json"))
-                .andExpect(MockMvcResultMatchers.content().string(CAR_JSON));
+                .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.content().string(
+                        objectMapper.writeValueAsString(AppTestUtil.carResponseDto)
+                ));
     }
 
-    private CarDto createCarDto() {
-        return new CarDto(1L, "Brand", "Color", "ABC123", "Model", 1, true);
+    @Test
+    @DisplayName("Test getCarById(Long); then returns NOT_FOUND status code")
+    void testGetCarById_thenReturnsNotFoundStatusCode_whenSuchCarDoesntExists() throws Exception {
+        //Arrange
+        doThrow(new CarNotFoundException("")).when(carService).getCarById(anyLong());
```

Comment:
Если не проверяется сообщение об ошибке, а только факт исключения, то лучше:
```suggestion
        doThrow(CarNotFoundException.class)
        .when(carService)
        .getCarById(anyLong());
```
А методы в цепочке вызовов следует размещать на отдельных строках.